railties/lib/rails/generators/rails/benchmark/benchmark_generator.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "rails/generators/named_base"
+
+module Rails
+  module Generators
+    class BenchmarkGenerator < NamedBase
+      argument :reports, type: :array, default: ["before", "after"]
+      class_option :method, type: :string, enum: %w(ips bm bmbm), default: "ips"
+
+      def generate_layout
+        if benchmark_method == "ips"
+          add_ips_to_gemfile unless ips_installed?
+        end
+
+        template("benchmark.rb.tt", "benchmarks/#{file_name}.rb")
+      end
+
+    private
+      def benchmark_method
+        options["method"]
+      end
+
+      def add_ips_to_gemfile
+        gem_group(:development, :test) do
+          gem(ips_gem_name)
+        end
+      end
+
+      def ips_installed?
+        in_root do
+          return File.read("Gemfile").match?(/gem.*\b#{ips_gem_name}\b.*/)
```

Comment:
Pretty surprised we don't already have a matcher for stuff like this, yet we have to read the Gemfile manually and custom write a regex? Seems odd. Does `gem` include something like this?

Reply:
Nope. Looks like @gmcgibbon had to add this for the change generator too: https://github.com/rails/rails/blob/f619ac91d33bd431d94726cc97092c3dbeb2afc6/railties/lib/rails/generators/rails/db/system/change/change_generator.rb#L53

Let's leave this for now, and potentially look at an extraction later.

Reply:
>  Let's leave this for now and potentially look at an extraction later.

@kaspth just for clarification, do you mean in a separate PR afterwards?